// diva-shared-apps/components/dropdown/dropdown.component.ts
/**
 * This file should contains all the logic and parameters related to dropdown.
 * Any changes in this file will lead to changes in all the places
 * where ever its refered
 */

import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { SelectItem } from 'primeng/api';

interface Dropdown {
  label: string;
  value: string;
}

@Component({
  selector: 'app-dropdown',
  templateUrl: './dropdown.component.html',
  styleUrls: ['./dropdown.component.css']
})
export class DropdownComponent implements OnInit {


  @Input() placeholder: string;
  @Input() label: string;
  @Input() showClear: boolean;
  @Input() selectedItem: string;
  @Output() selectedItemChange = new EventEmitter();
  filtereddatasource: any[] = [];
  originaldatasource: SelectItem[];
  @Input() size: number;
  @Input() disabled: boolean;
  @Input()
  set datasource (data: SelectItem[]) {
    console.log('data', data);
    this.filtereddatasource = [];
    this.originaldatasource = data;
    if (data && data.length > this.size) {
      for (let i = 0; i <= this.size; i++) {
        this.filtereddatasource.push(data[i]);
      }
    } else if (data) {
      this.filtereddatasource = data;
    }
    console.log('filtered Data', this.filtereddatasource);
  }

  get datasource () {
    return this.filtereddatasource;
  }
  constructor() {
    this.placeholder = 'Select option';
    this.size = 500;
  }

  ngOnInit() {
  }
  filterdropdown(event) {
    console.log('filterdropdown', event);
    this.filtereddatasource = [];
    if (event) {
      for (const item of this.originaldatasource) {
        if (item.label.toLowerCase().indexOf(event.toLowerCase()) === 0 && this.filtereddatasource.length < this.size) {
          this.filtereddatasource.push(item);
        } else if (this.filtereddatasource.length === this.size) {
          break;

        }
      }
    } else if (this.originaldatasource.length > this.size) {
      for (let i = 0; i <= this.size; i++) {
        this.filtereddatasource.push(this.originaldatasource[i]);
      }
    } else {
      this.filtereddatasource =  this.originaldatasource;
    }
    console.log('filteredcontrols', this.filtereddatasource);
  }
  change(data: any) {
    this.selectedItemChange.emit(data.value);
    console.log('change event', data.value);
  }
}
